### ISSUE TITLE
pytest does not work as intended

### ISSUE THREAD

**Issue by uvchik** [wind-python/windpowerlib#12]
...at least as I think it is intended :smirk: It is not urgent but if I want to contribute I need to run the tests.

I tried it on a originally cloned repository.

`pytest /path/to/local/git/windpowerlib/tests/` - works (collected 13 items)
`pytest /path/to/local/git/windpowerlib/windpowerlib/` - works (collected 7 items)

`pytest /path/to/local/git/windpowerlib/` - error (collected 22 items)

```python
=================================================================================================================== ERRORS ====================================================================================================================
__________________________________________________________________________________________________________ ERROR collecting setup.py __________________________________________________________________________________________________________
git_local/windpowerlib/setup.py:16: in <module>
    'pandas >= 0.13.1'])
.virtualenv/oemof_cosmos/lib/python3.5/site-packages/setuptools/__init__.py:129: in setup
    return distutils.core.setup(**attrs)
/usr/lib/python3.5/distutils/core.py:134: in setup
    ok = dist.parse_command_line()
.virtualenv/oemof_cosmos/lib/python3.5/site-packages/setuptools/dist.py:439: in parse_command_line
    result = _Distribution.parse_command_line(self)
/usr/lib/python3.5/distutils/dist.py:472: in parse_command_line
    args = self._parse_command_opts(parser, args)
.virtualenv/oemof_cosmos/lib/python3.5/site-packages/setuptools/dist.py:754: in _parse_command_opts
    nargs = _Distribution._parse_command_opts(self, parser, args)
/usr/lib/python3.5/distutils/dist.py:521: in _parse_command_opts
    raise SystemExit("invalid command name '%s'" % command)
E   SystemExit: invalid command name '/home/uwe/git_local/windpowerlib'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=========================================================================================================== 1 error in 0.81 seconds ===========================================================================================================
```

BTW: `pytest /path/to/local/git/oemof/`works.

**Comment by birgits** [comment]
I think it's a problem with the notebook test in the examples directory. I added an extras_require to the setup.py on [featrues/test_setup](https://github.com/wind-python/windpowerlib/tree/featrues/test_setup) branch that installs all extra packages needed for running the tests. If you install from this branch with

`pip install -e /path/to/windpowerlib/[test]`

does it work for you? It did for me in a fresh virtualenv.

**Comment by uvchik** [comment]
The problem above is solved and 21 test will be execute. 20 tests pass but the test_basic_example_ipynb raises an error. Here is the full output:

```python
============================================================================================================= test session starts =============================================================================================================
platform linux -- Python 3.5.3, pytest-3.4.2, py-1.5.2, pluggy-0.6.0
rootdir: /path/to/windpowerlib, inifile: pytest.ini
plugins: nose2pytest-1.0.8
collected 21 items                                                                                                                                                                                                                            

git_local/windpowerlib/example/test_examples.py .F                                                                                                                                                                                      [  9%]
git_local/windpowerlib/tests/test_density.py ..                                                                                                                                                                                         [ 19%]
git_local/windpowerlib/tests/test_modelchain.py ....                                                                                                                                                                                    [ 38%]
git_local/windpowerlib/tests/test_power_output.py ...                                                                                                                                                                                   [ 52%]
git_local/windpowerlib/tests/test_temperature.py .                                                                                                                                                                                      [ 57%]
git_local/windpowerlib/tests/test_tools.py .                                                                                                                                                                                            [ 61%]
git_local/windpowerlib/tests/test_wind_speed.py ..                                                                                                                                                                                      [ 71%]
git_local/windpowerlib/windpowerlib/modelchain.py ..                                                                                                                                                                                    [ 80%]
git_local/windpowerlib/windpowerlib/tools.py .                                                                                                                                                                                          [ 85%]
git_local/windpowerlib/windpowerlib/wind_turbine.py ...                                                                                                                                                                                 [100%]

================================================================================================================== FAILURES ===================================================================================================================
____________________________________________________________________________________________________ TestExamples.test_basic_example_ipynb ____________________________________________________________________________________________________

self = <example.test_examples.TestExamples object at 0x7f9c4c1f1748>

    def test_basic_example_ipynb(self):
        dir_path = os.path.dirname(os.path.realpath(__file__))
        nb, errors = self._notebook_run(os.path.join(dir_path,
>                                                    'basic_example.ipynb'))

test_examples.py:48: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
test_examples.py:37: in _notebook_run
    nb = nbformat.read(fout, nbformat.current_nbformat)
../../../.virtualenv/oemof_cosmos/lib/python3.5/site-packages/nbformat/__init__.py:141: in read
    return reads(fp.read(), as_version, **kwargs)
../../../.virtualenv/oemof_cosmos/lib/python3.5/site-packages/nbformat/__init__.py:74: in reads
    nb = reader.reads(s, **kwargs)
../../../.virtualenv/oemof_cosmos/lib/python3.5/site-packages/nbformat/reader.py:58: in reads
    nb_dict = parse_json(s, **kwargs)
../../../.virtualenv/oemof_cosmos/lib/python3.5/site-packages/nbformat/reader.py:14: in parse_json
    nb_dict = json.loads(s, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

s = b'{\n "cells": [\n  {\n   "cell_type": "markdown",\n   "metadata": {},\n   "source": [\n    "# Basic usage\\n",\n    "...ython",\n   "pygments_lexer": "ipython3",\n   "version": "3.4.2"\n  }\n },\n "nbformat": 4,\n "nbformat_minor": 1\n}\n'
encoding = None, cls = None, object_hook = None, parse_float = None, parse_int = None, parse_constant = None, object_pairs_hook = None, kw = {}

    def loads(s, encoding=None, cls=None, object_hook=None, parse_float=None,
            parse_int=None, parse_constant=None, object_pairs_hook=None, **kw):
        """Deserialize ``s`` (a ``str`` instance containing a JSON
        document) to a Python object.
    
        ``object_hook`` is an optional function that will be called with the
        result of any object literal decode (a ``dict``). The return value of
        ``object_hook`` will be used instead of the ``dict``. This feature
        can be used to implement custom decoders (e.g. JSON-RPC class hinting).
    
        ``object_pairs_hook`` is an optional function that will be called with the
        result of any object literal decoded with an ordered list of pairs.  The
        return value of ``object_pairs_hook`` will be used instead of the ``dict``.
        This feature can be used to implement custom decoders that rely on the
        order that the key and value pairs are decoded (for example,
        collections.OrderedDict will remember the order of insertion). If
        ``object_hook`` is also defined, the ``object_pairs_hook`` takes priority.
    
        ``parse_float``, if specified, will be called with the string
        of every JSON float to be decoded. By default this is equivalent to
        float(num_str). This can be used to use another datatype or parser
        for JSON floats (e.g. decimal.Decimal).
    
        ``parse_int``, if specified, will be called with the string
        of every JSON int to be decoded. By default this is equivalent to
        int(num_str). This can be used to use another datatype or parser
        for JSON integers (e.g. float).
    
        ``parse_constant``, if specified, will be called with one of the
        following strings: -Infinity, Infinity, NaN.
        This can be used to raise an exception if invalid JSON numbers
        are encountered.
    
        To use a custom ``JSONDecoder`` subclass, specify it with the ``cls``
        kwarg; otherwise ``JSONDecoder`` is used.
    
        The ``encoding`` argument is ignored and deprecated.
    
        """
        if not isinstance(s, str):
            raise TypeError('the JSON object must be str, not {!r}'.format(
>                               s.__class__.__name__))
E           TypeError: the JSON object must be str, not 'bytes'

/usr/lib/python3.5/json/__init__.py:312: TypeError
------------------------------------------------------------------------------------------------------------ Captured stderr call -------------------------------------------------------------------------------------------------------------
[NbConvertApp] Converting notebook /path/to/windpowerlib/example/basic_example.ipynb to notebook
[NbConvertApp] Executing notebook with kernel: python3
[NbConvertApp] Writing 79805 bytes to /tmp/tmpb2ma1qx3.ipynb
===================================================================================================== 1 failed, 20 passed in 9.90 seconds =====================================================================================================
```

**Comment by SabineHaas** [comment]
@uvchik am I right that you fixed this issue in #22? 

I am trying to clean up our issues to get an overview for a big release next year :)

**Comment by uvchik** [comment]
This one is solved. All tests have passed.
